ext/ForwardDiffExt.jl

```diff
@@ -45,21 +56,41 @@ end
 Wrap a log density that supports evaluation of `Value` to handle `ValueGradient`, using
 `ForwardDiff`.
 
-Keywords are passed on to `ForwardDiff.GradientConfig` to customize the setup. In
-particular, chunk size can be set with a `chunk` keyword argument (accepting an integer or a
-`ForwardDiff.Chunk`), and the underlying vector used by `ForwardDiff` can be set with the
-`x` keyword argument (accepting an `AbstractVector`).
+Keyword arguments:
+
+- `chunk` can be used to set the chunk size, an integer or a `ForwardDiff.Chunk`
+
+- `gradient_config_type` can be `nothing` (the default) or a type (eg `Float64`).
+
+   The latter preallocates and reuses a `ForwardDiff.GradientConfig` for that type. Note
+   that **this option is not thread-safe**. You can [`copy`](@ref) the results for
+   concurrent evaluation:
+   ```julia
+   ∇ℓ1 = ADgradient(:ForwardDiff, ℓ; gradient_config_type = Float64)
+   ∇ℓ2 = copy(∇ℓ1) # you can now use both, in different threads
+   ```
 """
 function ADgradient(::Val{:ForwardDiff}, ℓ;
                     x::Union{Nothing,AbstractVector} = nothing,
                     chunk::Union{Integer,ForwardDiff.Chunk} = _default_chunk(ℓ),
-                    gradientconfig::ForwardDiff.GradientConfig = _default_gradientconfig(ℓ, chunk, x))
-    ForwardDiffLogDensity(ℓ, gradientconfig)
+                    gradient_config_type::Union{Nothing,Type{T}} = nothing) where {T<:Real}
+    gradient_config = if gradient_config_type ≡ nothing
+        nothing
+    else
+        S = gradient_config_type
+        (isconcretetype(T) && (S <: Real)) ||
+            throw(ArgumentError("gradient_config_type needs to be a concrete subtype of Real."))
+        _make_gradient_config(S, ℓ, chunk)
```

Comment:
`S` is not needed, is it?
```suggestion
        isconcretetype(T) ||
            throw(ArgumentError("gradient_config_type needs to be a concrete subtype of Real."))
        _make_gradient_config(T, ℓ, chunk)
```